src/index.js

```diff
@@ -0,0 +1,73 @@
+const gridRows = 20;
+const gridCols = 10;
+const gameGrid = Array.from({ length: gridRows }, () => Array(gridCols).fill(0));
+
+const tetrisPiece = [ // creates all possible pieces and various colours
+    { shape: [[1, 1, 1, 1]], colour: 'blue' },
+    { shape: [[1, 1], [1, 1]], colour: 'yellow' },
+    { shape: [[1, 1, 1], [1, 0, 0]], colour: 'pink' },
+    { shape: [[1, 1, 1], [0, 0, 1]], colour: 'red' },
+    { shape: [[1, 1, 1], [0, 1, 0]], colour: 'green' },   
+    { shape: [[1, 1, 0], [0, 1, 1]], colour: 'maroon' },
+    { shape: [[0, 1, 1], [1, 1, 0]], colour: 'orange' }
+];
+
+function getrandompiece() { // creates a random piece and returns it
+    const randomInt = Math.floor(Math.random() * tetrisPiece.length);
+    return tetrisPiece[randomInt];
+}
+
+const randomPiece = getrandompiece(); // randomizes each piece and corresponding
+console.log("random tetris piece:", randomPiece.shape);
+console.log("random tetris colour:", randomPiece.colour);
+let randomPieceColour = randomPiece.colour;
+
+function placePiece(cols, rows, piece, gameGrid) { // places a piece in the game grid
+    for (let i = 0; i < piece.shape.length; i++) {
+        for (let j = 0; j < piece.shape[i].length; j++) {
+            if (piece.shape[i][j]) {
```

Comment:
In JavaScript, it is valid to use this form to check if a variable is not 0. Though, in most cases, it doesn't make the code much clearer, especially now as we're looking for specifically 1. It's probably best to change this to an explicit === 1
```suggestion
            if (piece.shape[i][j] === 1) {
```